tests/components/mochad/test_light.py

```diff
@@ -18,122 +16,57 @@ def pymochad_mock():
-
-
-class TestMochadLight256Levels(unittest.TestCase):
-    """Test for mochad light platform."""
-
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        controller_mock = mock.MagicMock()
-        dev_dict = {"address": "a1", "name": "fake_light", "brightness_levels": 256}
-        self.light = mochad.MochadLight(self.hass, controller_mock, dev_dict)
-
-    def teardown_method(self, method):
-        """Stop everything that was started."""
-        self.hass.stop()
-
-    def test_turn_on_with_no_brightness(self):
-        """Test turn_on."""
-        self.light.turn_on()
-        self.light.light.send_cmd.assert_called_once_with("xdim 255")
-
-    def test_turn_on_with_brightness(self):
-        """Test turn_on."""
-        self.light.turn_on(brightness=45)
-        self.light.light.send_cmd.assert_called_once_with("xdim 45")
-
-    def test_turn_off(self):
-        """Test turn_off."""
-        self.light.turn_off()
-        self.light.light.send_cmd.assert_called_once_with("off")
-
-
-class TestMochadLight64Levels(unittest.TestCase):
-    """Test for mochad light platform."""
-
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        controller_mock = mock.MagicMock()
-        dev_dict = {"address": "a1", "name": "fake_light", "brightness_levels": 64}
-        self.light = mochad.MochadLight(self.hass, controller_mock, dev_dict)
-
-    def teardown_method(self, method):
-        """Stop everything that was started."""
-        self.hass.stop()
-
-    def test_turn_on_with_no_brightness(self):
-        """Test turn_on."""
-        self.light.turn_on()
-        self.light.light.send_cmd.assert_called_once_with("xdim 63")
-
-    def test_turn_on_with_brightness(self):
-        """Test turn_on."""
-        self.light.turn_on(brightness=45)
-        self.light.light.send_cmd.assert_called_once_with("xdim 11")
-
-    def test_turn_off(self):
-        """Test turn_off."""
-        self.light.turn_off()
-        self.light.light.send_cmd.assert_called_once_with("off")
+@pytest.fixture
+def light_mock(hass, brightness):
+    """Mock light."""
+    controller_mock = mock.MagicMock()
+    dev_dict = {"address": "a1", "name": "fake_light", "brightness_levels": brightness}
+    return mochad.MochadLight(hass, controller_mock, dev_dict)
+
+
+async def test_setup_adds_proper_devices(hass):
+    """Test if setup adds devices."""
+    good_config = {
+        "mochad": {},
+        "light": {
+            "platform": "mochad",
+            "devices": [{"name": "Light1", "address": "a1"}],
+        },
+    }
+    assert await async_setup_component(hass, light.DOMAIN, good_config)
+
+
+@pytest.mark.parametrize("brightness", [32, 256, 64])
+async def test_name(light_mock):
+    """Test the name."""
+    assert "fake_light" == light_mock.name
+
+
+@pytest.mark.parametrize(
```

Comment:
This makes no sense either?

Reply:
I have combined the 3 old tests that have tested different light levels into one parametrized. The 3 old test have basically tested the same thing with different light levels. If this does not make sense we can replace these tests with more meaningful tests but I did not want to remove any test that was there before

Reply:
I am also confused why the test returns on with light level 32 and xdim with different light levels. However, this was also the case for the old tests. I would expect a more consistence behavior ..

Reply:
Interesting, I didn't realize that you could `parametrize` stuff in fixtures. TIL!